File: python/project_euler/problem_007/sol1.py
from math import sqrt

def is_prime(number: int) -> bool:
    
    if 1 < number < 4:
        
        return True
    elif number < 2 or number % 2 == 0 or number % 3 == 0:
        
        return False

    for i in range(5, int(sqrt(number) + 1), 6):
        if number % i == 0 or number % (i + 2) == 0:
            return False
    return True

def solution(nth: int = 10001) -> int:
    
    count = 0
    number = 1
    while count != nth and number < 3:
        number += 1
        if is_prime(number):
            count += 1
    while count != nth:
        number += 2
        if is_prime(number):
            count += 1
    return number

if __name__ == "__main__":
    print(f"{solution() = }")

File: ruby/project_euler/problem_007/sol1.rb
def is_prime?(number)   
    value = true
    if number > 1 and number < 4
        
        value = true
    elsif number < 2 or number % 2 == 0 or number % 3 == 0
    
        value = false
    end
    end_range = (Math.sqrt(number) + 1).to_i
    
    for i in (5..end_range).step(6)
        if number % i == 0 or number % (i + 2) == 0
            value = false
        end
    end
    result = value
end

def solution(nth = 10001)
  primes = Array.new()
  num = 2
  while primes.length < nth
    if is_prime?(num)
      primes.append(num)
    end
    num += 1
  end
  primes[primes.length - 1]
end

answer = solution()
p answer